ext/spl/spl_iterators.c

```diff
@@ -1963,21 +1984,26 @@ PHP_METHOD(RecursiveRegexIterator, getChildren)
 	SPL_FETCH_AND_CHECK_DUAL_IT(intern, ZEND_THIS);
 
 	zend_call_method_with_0_params(Z_OBJ(intern->inner.zobject), intern->inner.ce, NULL, "getchildren", &retval);
-	if (!EG(exception)) {
-		zval args[5];
+	if (EG(exception)) {
+		zval_ptr_dtor(&retval);
+		RETURN_THROWS();
+	}
 
-		ZVAL_COPY(&args[0], &retval);
-		ZVAL_STR_COPY(&args[1], intern->u.regex.regex);
-		ZVAL_LONG(&args[2], intern->u.regex.mode);
-		ZVAL_LONG(&args[3], intern->u.regex.flags);
-		ZVAL_LONG(&args[4], intern->u.regex.preg_flags);
+	zval args[5];
+	args[0] = retval;
```

Comment:
Why are you assigning here instead of using the copy macro?
Note that an assignment copies the "extra" fields too

Reply:
Because I thought I could, but I'll use the macro then.